include/libcyphal/transport/defines.hpp

```diff
@@ -28,11 +30,71 @@ using TransferId = std::uint64_t;
 
 struct ProtocolParams final
 {
-    NodeId        max_nodes;
-    std::size_t   mtu_bytes;
-    std::uint64_t transfer_id_modulo;
+    TransferId  transfer_id_modulo;
+    std::size_t mtu_bytes;
+    NodeId      max_nodes;
+
+    ProtocolParams(TransferId _transfer_id_modulo, std::size_t _mtu_bytes, NodeId _max_nodes)
+        : transfer_id_modulo{_transfer_id_modulo}
+        , mtu_bytes{_mtu_bytes}
+        , max_nodes{_max_nodes}
+    {
+    }
+};
+
+struct TransferMetadata
+{
+    TransferId transfer_id;
+    TimePoint  timestamp;
+    Priority   priority;
+
+    TransferMetadata(TransferId _transfer_id, TimePoint _timestamp, Priority _priority)
+        : transfer_id{_transfer_id}
+        , timestamp{_timestamp}
+        , priority{_priority}
+    {
+    }
+};
+
+struct ServiceTransferMetadata final : TransferMetadata
+{
+    NodeId remote_node_id;
+
+    ServiceTransferMetadata(TransferId _transfer_id, TimePoint _timestamp, Priority _priority, NodeId _remote_node_id)
+        : TransferMetadata{_transfer_id, _timestamp, _priority}
+        , remote_node_id{_remote_node_id}
+    {
+    }
+};
+
+// TODO: Maybe have `cetl::byte` polyfill for C++20
+using PayloadFragments = cetl::span<cetl::span<uint8_t>>;
```

Comment:
The C++ standard does not define `::uint8_t`

```suggestion
using PayloadFragments = cetl::span<cetl::span<std::uint8_t>>;
```

Reply:
that was typo